src/randoop/operation/ConstructorSignatures.java

```diff
@@ -0,0 +1,73 @@
+package randoop.operation;
+
+import java.lang.reflect.Constructor;
+
+/**
+ * ConstructorParser provides static methods to write as well as parse 
+ * a string representation of a constructor signature.
+ */
+public class ConstructorSignatures {
+
+  /**
+   * Parses a constructor signature as produced by 
+   * {@link ConstructorSignatures#getSignatureString(Constructor)} 
+   * and returns the corresponding reflective {@link Constructor} object.
+   * 
+   * @param signature a string representing a constructor signature.
+   * @return reflective {@link Constructor} method corresponding to signature.
+   * @throws OperationParseException if signature parameter does not match 
+   *         expected format.
+   */
+  public static Constructor<?> getConstructorForSignatureString(String signature) throws OperationParseException {
+    if (signature == null) {
+      throw new IllegalArgumentException("signature may not be null");
+    }
+    
+    int openPar = signature.indexOf('(');
```

Comment:
Would matching against a single regular expression be easier to read than this sequence of character operations?